src/test/java/org/junitpioneer/jupiter/RangeSourceTests.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2015-2018 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.platform.commons.util.PreconditionViolationException;
+
+/**
+ * A test for the {@link RangeSource} annotation and the underlying {@link RangeSourceProvider}.
+ */
+public class RangeSourceTests {
+	private static final List<Integer> allValues = new ArrayList<>();
+
+	/**
+	 * Fill up {@code allValues} with some values.
+	 * These values will be removed as {@link #ascending(int)} and {@link #descending(int)} receive their arguments,
+	 * and at the end of the test suite, {@link #assertAllValuesSupplied()} will check that all the values were
+	 * properly provided.
```

Comment:
Hah, this is very creative. :D

I'd still prefer it a little more formal, though. Can you have a look at, for example, [`DefaultLocaleTests`](https://github.com/junit-pioneer/junit-pioneer/blob/master/src/test/java/org/junitpioneer/jupiter/DefaultLocaleTests.java#L31) to see how to test extensions? It follows these steps:

* create an inner test class with the setup you want to test; these look very similar to how the user's tests will eventually look
* in _your_ tests, ask `AbstractPioneerTestEngineTests::executeTestsForClass` to execute these inner test classes
* assert proper execution

This includes misconfigurations.